actionpack/lib/action_controller.rb

```diff
@@ -54,6 +54,12 @@ module ActionController
   autoload :TemplateAssertions, "action_controller/test_case"
 end
 
+ActiveSupport.on_load(:after_initialize) do
```

Comment:
It would be way more clean to put it into `ActionController#eager_load!`, however when that one is invoked the app controllers itself are not loaded yet.

I thought that maybe calling `ActiveSupport.on_load` inline from the root component class is shady but then I found other components doing it too: https://github.com/rails/rails/blob/master/actionpack/lib/action_dispatch.rb#L106